api/datastore/sql/migrations/migrations.go

```
@@ -1,534 +1,136 @@
-	return bindataRead(
-		__7_add_route_cpusUpSql,
-		"7_add_route_cpus.up.sql",
-	)
+	return migrations
 }
 
-func _7_add_route_cpusUpSql() (*asset, error) {
-	bytes, err := _7_add_route_cpusUpSqlBytes()
-	if err != nil {
-		return nil, err
-	}
+func DownAll(driver string, db *sql.DB) error {
+	goose.SetDialect(driver)
+	migrations = sortAndConnectMigrations(migrations)
 
-	info := bindataFileInfo{name: "7_add_route_cpus.up.sql", size: 33, mode: os.FileMode(420), modTime: time.Unix(1515628068, 0)}
-	a := &asset{bytes: bytes, info: info}
-	return a, nil
-}
-
-// Asset loads and returns the asset for the given name.
-// It returns an error if the asset could not be found or
-// could not be loaded.
-func Asset(name string) ([]byte, error) {
-	cannonicalName := strings.Replace(name, "\\", "/", -1)
-	if f, ok := _bindata[cannonicalName]; ok {
-		a, err := f()
+	for {
+		currentVersion, err := goose.GetDBVersion(db)
 		if err != nil {
-			return nil, fmt.Errorf("Asset %s can't read by error: %v", name, err)
+			return err
 		}
-		return a.bytes, nil
-	}
-	return nil, fmt.Errorf("Asset %s not found", name)
-}
 
-// MustAsset is like Asset but panics when Asset would return an error.
-// It simplifies safe initialization of global variables.
-func MustAsset(name string) []byte {
-	a, err := Asset(name)
-	if err != nil {
-		panic("asset: Asset(" + name + "): " + err.Error())
-	}
-
-	return a
-}
-
-// AssetInfo loads and returns the asset info for the given name.
-// It returns an error if the asset could not be found or
-// could not be loaded.
-func AssetInfo(name string) (os.FileInfo, error) {
-	cannonicalName := strings.Replace(name, "\\", "/", -1)
-	if f, ok := _bindata[cannonicalName]; ok {
-		a, err := f()
+		current, err := migrations.Current(currentVersion)
 		if err != nil {
-			return nil, fmt.Errorf("AssetInfo %s can't read by error: %v", name, err)
+			fmt.Printf("goose: no migrations to run. current version: %d\n", currentVersion)
+			return nil
```

should we check for a specific error here? should we return this error? unclear if `return nil` is in err

---

fixex